### DISCUSSION TITLE
How to change the input type of the 2FA code on the login page

### DISCUSSION THREAD

**Discussion by t-rutter** [gravitational/teleport#10389] [Q&A] [answered]
I've found that on the login page if the 'type' attribute for the 2FA input field is changed from 'text' to 'tel' Safari will allow autofill of the 2FA code.

Where can this change be made ?
 I've tried to search through the code to no avail

Change the html on the login page
From:
`<input type="text" placeholder="123 456" autocomplete="off" color="text.onLight" class="sc-fzozJi kFXnqq" value="">`
To:
`<input type="tel" placeholder="123 456" autocomplete="off" color="text.onLight" class="sc-fzozJi kFXnqq" value="">`

**Comment by Valien** [comment, marked as answer]
@t-rutter - you can check this repository: https://github.com/gravitational/webapps - I believe that is the one that handles the Web UI assets. You can also open up a GitHub feature request/issue to submit this suggestion as well.

**Reply by t-rutter** [reply]
Hi @Valien 
I think I found where in the source the form comes from 'FormLogin.tsx'.
I've created a pull request adding a single line 'type="tel"' to the relevent section in 'FormLogin.tsx'.
Not quite sure how I go about testing this project, So if someone from the team could please test 😄 
https://github.com/gravitational/webapps/pull/608

**Reply by Valien** [reply]
Hey @t-rutter looks like your PR is in review process! Thanks for contributing to make Teleport a better tool!

**Reply by t-rutter** [reply]
Hi @Valien, my PR just got merged into master. 🎉 
Thank you to the Gravitational team and all contributors for this great tool! 👍

**Reply by Valien** [reply]
That is awesome!
